src/amo/reducers/versions.js

```diff
@@ -407,6 +407,12 @@ const reducer = (
     case SEARCH_LOADED: {
       const { addons, addonsResponse, results } = action.payload;
 
+      // We might have no add-ons at all, e.g., after a collection has been
+      // created.
+      if (!addons && !addonsResponse && !results) {
```

Comment:
We'd return almost the same thing, but avoid duplication

Reply:
I figured it was better to shortcut the whole function and just return if we know there are no add-ons. I could do something like check for `addons` being falsey in the final `else`, and then set `items` to an empty array, which would accomplish the same thing, but then we are still executing some extra code (albeit very little) that isn't needed. And I also think it will be a bit ugly with an extra `if` in there. If you feel that is better then I can do it, but I actually prefer what I have here.

Reply:
I don't have strong opinion on that. Let's try it your way.